ui/src/components/content/catalog/services/ServiceProvider.tsx

```diff
@@ -0,0 +1,105 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * SPDX-FileCopyrightText: Huawei Inc.
+ */
+
+import React, { useEffect, useState } from 'react';
+import { Tabs } from 'antd';
+import ServiceDetail from './ServiceDetail';
+import { useEmotionCss } from '@ant-design/use-emotion-css';
+import { CategoryOclVo, OclDetailVo, ProviderOclVo, VersionOclVo } from '../../../../xpanse-api/generated';
+import { Tab } from 'rc-tabs/lib/interface';
+
+let lastServiceName: string = '';
+
+function ServiceProvider({
+    categoryOclData,
+    serviceName,
+}: {
+    categoryOclData: CategoryOclVo[];
+    serviceName: string;
+}): JSX.Element {
+    const [activeKey, setActiveKey] = useState<string>('');
+    const [serviceDetails, setServiceDetails] = useState<OclDetailVo>(new OclDetailVo());
+    const [serviceRegions, setServiceRegions] = useState<string[]>([]);
+
+    const detailMapper: Map<string, OclDetailVo> = new Map<string, OclDetailVo>();
+    const regionMapper: Map<string, string[]> = new Map<string, string[]>();
+    const [name, version] = serviceName.split('@');
+
+    const items: Tab[] = categoryOclData
+        .filter((service: CategoryOclVo) => service.name === name)
+        .flatMap((service: CategoryOclVo) => service.versions)
+        .filter((v) => (v as VersionOclVo).version === version)
+        .flatMap((v) => {
+            if (!v || !v.cloudProvider) {
+                return { key: '', label: '' };
+            }
+            return v.cloudProvider.map((cloudProvider: ProviderOclVo) => {
+                if (!cloudProvider.details) {
+                    return { key: '', label: '' };
+                }
+                const key = serviceName + '@' + cloudProvider.name;
+                detailMapper.set(key, cloudProvider.details[0]);
+                regionMapper.set(key, cloudProvider.regions || []);
+                const name = cloudProvider.name!.toString();
+                return {
+                    label: name,
+                    key: name,
+                    children: [],
+                };
+            });
+        });
+
+    useEffect(() => {
+        if (items.length > 0 && lastServiceName !== serviceName) {
+            const key = serviceName + '@' + items[0]!.key;
```

Comment:
does the TS compiler here complain about items[0] being null? if not, we can remove the null assertion here `!`

Reply:
fixed.